src/python/director/drakevisualizer.py

```diff
@@ -215,62 +240,68 @@ def loadPolyDataMeshes(geom):
             print 'warning, cannot find file:', filename
             return []
 
+        mesh_vis_list = []
+
         if filename.endswith('vtm'):
             polyDataList = ioUtils.readMultiBlock(filename)
+            mesh_vis_list = Geometry.DeriveMeshVisList(geom = geom, \
+                meshFileName = filename, polyDataList = polyDataList)
         else:
-            polyDataList = [ioUtils.readPolyData(filename)]
-
-        if USE_TEXTURE_MESHES:
-            for polyData in polyDataList:
-                Geometry.loadTextureForMesh(polyData, filename)
-
-
-        Geometry.MeshCache[filename] = list(polyDataList)
-
-        return polyDataList
+            if os.path.isfile(basename + '.mtl'):
+                polyDataList, actorList = ioUtils.readObjMtl(geom.string_data)
+                for polyData, actor in zip(polyDataList, actorList):
+                    color = actor.GetProperty().GetColor()
+                    alpha = actor.GetProperty().GetOpacity()
+                    texture = actor.GetTexture()
+                    mesh_vis = FieldContainer( \
+                        polyData = polyData, color = color, alpha = alpha, \
+                        texture = texture)
+                    mesh_vis_list.append(mesh_vis);
+            else:
+                polyDataList = [ioUtils.readPolyData(filename)]
+                mesh_vis_list = Geometry.DeriveMeshVisList(geom = geom, \
```

Comment:
style: don't use `name=arg` style arg list, just a regular positional arg list.  this comment applies to other areas in the diff, too